File: bitget-golang-sdk-api/pkg/model/mix/order/placeorderreq.go
package order

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: place an order request
 */
type PlaceOrderReq struct {
	/**
	 * Currency pair
	 */
	Symbol string `json:"symbol"`
	/**
	 * Deposit currency
	 */
	MarginCoin string `json:"marginCoin"`
	/**
	 * Open more, open more, empty more, empty more
	 */
	Side string `json:"side"`
	/**
	 * Client ID
	 */
	ClientOid string `json:"clientOid"`
	/**
	 * Amount of currency placed
	 */
	Size string `json:"size"`
	/**
	 * Order Type Market Price Limit
	 */
	OrderType string `json:"orderType"`
	/**
	 * Entrusted price
	 */
	Price string `json:"price"`
	/**
	 * Order validity
	 */
	TimeInForceValue string `json:"timeInForceValue"`
	/**
	 * Default stop profit price
	 */
	PresetTakeProfitPrice string `json:"presetTakeProfitPrice"`
	/**
	 * Preset stop loss price
	 */
	PresetStopLossPrice string `json:"presetStopLossPrice"`
}

File: bitget-node-sdk-api/src/lib/model/mix/order/PlaceOrderReq.ts
export class PlaceOrderReq{
    /**
     * Currency pair
     */
    private _symbol!:string;
    /**
     * Deposit currency
     */
    private _marginCoin!:string;
    /**
     * Client ID
     */
    private _clientOid!:string;
    /**
     * Amount of currency placed
     */
    private _size!:string;
    /**
     * Open more, open more, empty more, empty more
     */
    private _side!:string;
    /**
     * Order Type Market Price Limit
     */
    private _orderType!:string;
    /**
     * Entrusted price
     */
    private _price!:string;
    /**
     * Order validity
     */
    private _timeInForceValue!:string;
    /**
     * Default stop profit price
     */
    private _presetTakeProfitPrice!:string;
    /**
     * Preset stop loss price
     */
    private _presetStopLossPrice!:string;


    get symbol(): string {
        return this._symbol;
    }

    set symbol(value: string) {
        this._symbol = value;
    }

    get marginCoin(): string {
        return this._marginCoin;
    }

    set marginCoin(value: string) {
        this._marginCoin = value;
    }

    get clientOid(): string {
        return this._clientOid;
    }

    set clientOid(value: string) {
        this._clientOid = value;
    }

    get size(): string {
        return this._size;
    }

    set size(value: string) {
        this._size = value;
    }

    get side(): string {
        return this._side;
    }

    set side(value: string) {
        this._side = value;
    }

    get orderType(): string {
        return this._orderType;
    }

    set orderType(value: string) {
        this._orderType = value;
    }

    get price(): string {
        return this._price;
    }

    set price(value: string) {
        this._price = value;
    }

    get timeInForceValue(): string {
        return this._timeInForceValue;
    }

    set timeInForceValue(value: string) {
        this._timeInForceValue = value;
    }

    get presetTakeProfitPrice(): string {
        return this._presetTakeProfitPrice;
    }

    set presetTakeProfitPrice(value: string) {
        this._presetTakeProfitPrice = value;
    }

    get presetStopLossPrice(): string {
        return this._presetStopLossPrice;
    }

    set presetStopLossPrice(value: string) {
        this._presetStopLossPrice = value;
    }
}